script: Update `media` to include fix for `AudioBufferSourceNode.loop`

Taym95 opened the pull request:
Fix in https://github.com/servo/media/pull/476:

Previously:
the moment we set bufferSource.loop = true, the node hit the early “extreme edge case” check, decided “I refuse to output”, triggered onended, and returned silence. So the loop never got a chance to run.

Now:

Early bailout is removed(no idea why was there in first place).
The per tick offset is guaranteed to be a sensible value with respect to the loop range (wrapped if extreme).
The normal mixing code runs, updating pos each tick and respecting the loop region

Testing: I tested this manually, it would be nice to add a test for it, and I’m surprised there isn’t one yet.
Fixes: #41073

mrobinson commented:
Looks like this is still a draft and doesn't include the changes, but please include a description of the fix made in in the commit description and title. The motivation is this: if we end up having the fix `bufferSource.loop` more than once, a title describing what exactly was fixed will allow someone reading the commit log to understand the difference between the two:

From  https://book.servo.org/pr-description.html:
>  Try to avoid generic verbs such as "fix", "correct", or "improve" and instead describe what the fix does. Code can be fixed multiple times, but the message should more uniquely identify the change.

 Thank you!

Taym95 committed:
Fix bufferSource.loop = true on extreme edge case

Previously:
the moment we set bufferSource.loop = true, the node hit the early “extreme edge case” check, decided “I refuse to output”, triggered onended, and returned silence. So the loop never got a chance to run.

Now:

Early bailout is removed(no idea why was there in first place).
The per tick offset is guaranteed to be a sensible value with respect to the loop range (wrapped if extreme).
The normal mixing code runs, updating pos each tick and respecting the loop region

Signed-off-by: Taym Haddadi <haddadi.taym@gmail.com>

Taym95 committed:
Add "Looped AudioBufferSourceNode Keeps Rendering" test

Signed-off-by: Taym Haddadi <haddadi.taym@gmail.com>

servo-wpt-sync commented:
🤖 Opened new upstream WPT pull request (web-platform-tests/wpt#56621) with upstreamable changes.

jdm reviewed (approved):
Nice test!
